packages/swapper/src/manager/SwapperManager.ts

```diff
@@ -92,25 +94,33 @@ export class SwapperManager {
       buyAssetId: buyAsset.assetId,
     })
 
-    const settledSwapperDetailRequests: PromiseSettledResult<SwapperWithQuoteMetadata>[] =
-      await Promise.allSettled(
-        supportedSwappers.map(async swapper => {
-          const quote = await swapper.getTradeQuote(args)
-          const ratio = await getRatioFromQuote(quote, swapper, feeAsset)
-
-          return {
-            swapper,
-            quote,
-            inputOutputRatio: ratio,
-          }
-        }),
-      )
+    const settledSwapperDetailRequests: PromiseSettledResult<
+      Result<SwapperWithQuoteMetadata, SwapErrorMonad>
+      // TODO(gomes): do we still need allSettled? Can we use Promise.all? This should never reject anymore
+      // as long as we scutinize the error-handling flow
+    >[] = await Promise.allSettled(
+      supportedSwappers.map(async swapper => {
+        const maybeQuote = await swapper.getTradeQuote(args)
+
+        if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
+        const quote = maybeQuote.unwrap()
+
+        const ratio = await getRatioFromQuote(quote, swapper, feeAsset)
+
+        return Ok({
+          swapper,
+          quote,
+          inputOutputRatio: ratio,
+        })
+      }),
+    )
 
```

Comment:
I've to propose something along these lines:
```
    const settledSwapperDetailRequests: PromiseSettledResult<
      Result<SwapperWithQuoteMetadata, SwapErrorMonad>
      // TODO(gomes): do we still need allSettled? Can we use Promise.all? This should never reject anymore
      // as long as we scutinize the error-handling flow
    >[] = await Promise.allSettled(
      supportedSwappers.map(async swapper => {
        const maybeQuote = await swapper.getTradeQuote(args)
        const dd = maybeQuote.map(async quote => 
          {
            const ratio = await getRatioFromQuote(quote, swapper, feeAsset)
            return Ok({
              swapper,
              quote,
              inputOutputRatio: ratio,
            })
          }).mapErr(_ => Err(maybeQuote.unwrapErr))
        return dd
      })
    )
    ``` but yeah there's some type issues, I'm good with this for now